scripts/elemental-airgap.sh

```diff
@@ -0,0 +1,219 @@
+    [-c|--chart-repo path] source chart repo to pull elemental charts from (without the chart names).
+    [-l|--image-list path] generated text file with the list of saved images (one image per line).
+    [-i|--images path] tar.gz gernerated by docker save.
+    [-d|--debug] enable debug output on screen.
+    [-h|--help] Usage message.
+
+    Parameters could also be set passing env vars:
+    EOC_REGISTRY (-c)               : $EOC_REGISTRY
+    EOC_OPERATOR_CHART              : $EOC_OPERATOR_CHART
+    EOC_CRDS_CHART                  : $EOC_CRDS_CHART
+    CONTAINER_IMAGES_NAME           : $CONTAINER_IMAGES_NAME
+    CONTAINER_IMAGES_FILE (-l)      : $CONTAINER_IMAGES_FILE
+    CONTAINER_IMAGES_ARCHIVE (-i)   : $CONTAINER_IMAGES_ARCHIVE
+    DEBUG (-d)                      : $DEBUG
+    LOCAL_REGISTRY                  : $LOCAL_REGISTRY
+
+EOF
+}
+
+parse_parameters() {
+
+    local help
+
+    POSITIONAL=()
+    while [[ $# -gt 0 ]]; do
+        key="$1"
+        case $key in
+            -i|--images)
+            CONTAINER_IMAGES_ARCHIVE="$2"
+            shift # past argument
+            shift # past value
+            ;;
+            -l|--image-list)
+            CONTAINER_IMAGES_FILE="$2"
+            shift # past argument
+            shift # past value
+            ;;
+            -s|--source-registry)
+            EOC_REGISTRY="$2"
+            shift # past argument
+            shift # past value
+            ;;
+            -d|--debug)
+            DEBUG="true"
+            shift
+            ;;
+            -h|--help)
+            help="true"
+            shift
+            ;;
+            *)
+            print_help
+            exit 1
+            ;;
+        esac
+    done
+
+    if [ "$help" = "true" ]; then
+        print_help
+        exit 0
+    fi
+}
+
+exit_error() {
+    eval msg=\"$1\"
+    echo -e "ERR: $msg"
+    exit 1   
+}
+
+log_debug() {
+    [ "$DEBUG" = "false" ] && return
+    eval msg=\"$1\"
+    echo -e "$msg"
+}
+
+log_info() {
+    eval msg=\"$1\"
+    echo -e "$msg"
+}
+
+get_chart_val() {
+    var="$1"
+    val="$2"
+
+    eval $var=$(helm show values $CHART_NAME_OPERATOR | eval yq eval '.${val}' | sed s/\"//g 2>&1)
+    eval log_debug \"extracted $var\\t: \$$var\ \($val\)\"
+}
+
+prereq_checks() {
+    log_debug "Check required binaries availability"
+    for cmd in helm yq sed docker; do
+        if ! command -v "$cmd" > /dev/null; then
+            exit_error "'$cmd' not found."
+        fi
+        log_debug "$cmd found"
+    done
+}
+
+fetch_charts() {
+    local outstr chart
+
+    log_debug "fetching chart '$EOC_OPERATOR_CHART'"
+    if ! outstr=$(helm pull ${EOC_OPERATOR_CHART} 2>&1); then
+        exit_error "downloading ${EOC_OPERATOR_CHART}:\n $outstr"
+    fi
+    CHART_NAME_OPERATOR=$(ls -t1 | head -n 1)
+    log_info "Downloaded Elemental Operator chart: $CHART_NAME_OPERATOR"
+
+    log_debug "fetching chart '$EOC_CRDS_CHART'"
+    if ! outstr=$(helm pull ${EOC_CRDS_CHART} 2>&1); then
```

Comment:
Does it work for an OCI repository? I don't think it works without an specific version. Anyway, if needed we can easily publish the old charts in gh-pages and start updating the docs to use that helm repo.

Reply:
yep it works with an oci registry, but it will pick-up the latest tag.
While the idea to download the chart for the user is nice, it could cause pain if we force it (and with tags!).
Ideally we may want the user to pass the elemental-operator chart as an already downloaded archive... or as a link , which will be downloaded.
Ok, pushing a commit on top of this PR.

Reply:
Ok, now the elemental-operator chart is a mandatory argument. It could be:
1) a URL (https or oci): for https, means a link to a tar.gz btw.
2) a file: it expects a tgz elemental-operator chart one, will exit anyway if it will not be an elemental-operator chart
3) "dev" or "staging": it will automatically pick the latest dev or staging charts. Also stable is supported there but will fail as current stable doesn't support airgap (misses the registryUrl templating)